### ISSUE TITLE
Introduction

### ISSUE THREAD

**Issue by sksamuel** [kotlintest/kommons#1]
You interested in creating a kommons library in here @helmbold

**Comment by helmbold** [comment]
Yes, I'm interested. I think this is a good place to put common functionality of other projects. It could grow by practical needs.

**Comment by sksamuel** [comment]
The duration stuff is a good start. I want to add options, trys, and eithers. I use those all the time in Scala and the ? operators just aren't the same. Options give you nice composibility for example. Then there's validation constructs.

**Comment by helmbold** [comment]
I've used the "functional java" library, which tried to bring good ideas from Scala to Java and it was an overall bad experience! The code looked strange and the performance was poor. I think that the JetBrains guys had good reasons to go with "nullable" instead of `Option`. I like the mentioned constructs known from Scala, but I'm not sure whether I want them in Kotlin (still thinking ;- ) ).

**Comment by sksamuel** [comment]
I can see your point and I do use some nullable stuff, but things like this

`option.orElse(option2).orElse(option3).getOrElse(default)`
`Try { some code }.toOption().getOrElse(default)`

Are really nice. I think the null safety stuff in Kotlin is not as nice as the optionals in Java and Scala.
